src/group_impl.h

```diff
@@ -241,7 +241,7 @@ static int secp256k1_gej_eq_x_var(const secp256k1_fe *x, const secp256k1_gej *a)
     secp256k1_fe r, r2;
     VERIFY_CHECK(!a->infinity);
     secp256k1_fe_sqr(&r, &a->z); secp256k1_fe_mul(&r, &r, x);
-    r2 = a->x; secp256k1_fe_normalize_weak(&r2);
+    r2 = a->x;
```

Comment:
The `r2` variable isn't needed anymore; you can just call `return secp256k1_fe_equal_var(&r, &a->x);` directly.

Reply:
Nice catch, Thanks!